ci-operator/step-registry/cucushift/installer/check/azure/des/cucushift-installer-check-azure-des-commands.sh

```diff
@@ -0,0 +1,97 @@
+echo "OCP Version: $ocp_version"
+ocp_minor_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $2}' )
+
+# set the parameters we'll need as env vars
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+
+# log in with az
+if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+    az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURESTACK_ENDPOINT=$(cat "${SHARED_DIR}"/AZURESTACK_ENDPOINT)
+    SUFFIX_ENDPOINT=$(cat "${SHARED_DIR}"/SUFFIX_ENDPOINT)
+
+    if [[ -f "${CLUSTER_PROFILE_DIR}/ca.pem" ]]; then
+        cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID "${SHARED_DIR}"/metadata.json)
+CLUSTER_RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${CLUSTER_RESOURCE_GROUP}" ]]; then
+    CLUSTER_RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+DES_RESOURCE_GROUP=$(< "${SHARED_DIR}/resourcegroup")
+DES_NAME=$(< "${SHARED_DIR}/azure_des")
+
+critical_check_result=0
+
+#Get des id
+des_id=$(az disk-encryption-set show -n "${DES_NAME}" -g "${DES_RESOURCE_GROUP}" --query '[id]' -otsv)
+
+#check that node os disk is encrypted
+nodes_list=$(oc get nodes --no-headers | awk '{print $1}')
+for node in ${nodes_list}; do
+    echo "--- check node ${node} ---"
+    node_des_id=$(az vm show --name "${node}" -g "${CLUSTER_RESOURCE_GROUP}" --query 'storageProfile.osDisk.managedDisk.diskEncryptionSet.id' -otsv)
+    if [[ "${node_des_id}" == "${des_id}" ]]; then
+        echo "INFO: os disk for node ${node} is encrypted"
+    else
+        echo "ERROR: Get unexpected des id on os disk for node ${node}! expected value: ${des_id}, real value: ${node_des_id}"
+        critical_check_result=1 
+    fi
+done
+
+#check des setting in default sc
+echo "--- check des setting in default sc ---"
+if (( ocp_minor_version < 13 )) || [[ "${ENABLE_DES_DEFAULT_MACHINE}" != "true" ]]; then
```

Comment:
Shall we take `PERSISTENT_MONITORING == false` into consideration?

Reply:
I guess no need, ENV `PERSISTENT_MONITORING` only impacts whether to configure pv for prometheus data, it is unrelated with des setting in storage class.

And this step is added into specific workflow `cucushift-installer-rehearse-azure-ipi-disk-encryption-set-provision-chain.yaml` which does not set ENV `PERSISTENT_MONITORING`, and we also need step `ipi-install-monitoringpvc` to configure pv for prometheus to make sure that there is no permission issues.

Reply:
okay.